stan/math/rev/core/operator_minus_equal.hpp

```diff
@@ -8,13 +8,16 @@
 namespace stan {
 namespace math {
 
-inline var& var::operator-=(var b) {
+template <typename T>
+inline var_value<T>& var_value<T, require_floating_point_t<T>>::operator-=(
+    const var_value<T>& b) {
   vi_ = new internal::subtract_vv_vari(vi_, b.vi_);
   return *this;
 }
 
-template <typename Arith, require_arithmetic_t<Arith>*>
-inline var& var::operator-=(Arith b) {
+template <typename T>
+inline var_value<T>& var_value<T, require_floating_point_t<T>>::operator-=(
```

Comment:
The Arith stuff isn't necessary here because something like:

```
var a(5.0);
a -= 1;
```

will automatically promote the int to a double?

Reply:
Huh, that might be a problem. I think it will try to instantiate the function with T=int, which require prevents. I think those need to be require_arithmetic (and we might need some tests for doing arithmetic with vars and ints).

Reply:
`require_arithmetic` is not necessary here because of the require condition on `T`

```cpp
inline var_value<T>& var_value<T, require_floating_point_t<T>>::operator-=(T b)
```

Since `T` can only be floating point ints will be coalesced to floating point. The problem with not having that is both

1. Can't do partial specializations as mentioned above
2. When we allow `internal::subtract_vd_vari(vi_, b);` to be `internal::subtract_vari<var_value<T>, T>(vi_, b);` We want to make sure the non-var side `T` is a floating point type

I originally allowed for ints there through some weird hocus pocus but imo I agree'd with the change to just support floating point in var because it makes things a lot simpler

Reply:
I think coalescing to type is not c++ terminology, so I'm not sure what you mean by that. Anyway, I think this won't compile as compiler won't deduce template and do implicit conversion on same argument. You can try to prove me wrong by adding a test for this :).

Reply:
> Anyway, I think this won't compile as compiler won't deduce template and do implicit conversion on same argument.

Oh that explains the necessity of the `is_convertible` logic elsewhere. That's inconvenient.

> I agree'd with the change to just support floating point in var

I may have f'ed up telling you that. We do want:

```
var a(5.0);
a += 1;
```

to work.

Reply:
But wouldn't the template types be known at this point? The compiler wouldn't be doing template deduction and implicit conversion at the same time. That is the class template, not a separate function template.

Reply:
Of course, I was wrong. This is fine.